Sources/OTel/Metrics/SwiftMetricsFactory/OTLPMetricsFactory.swift

```diff
@@ -0,0 +1,190 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CoreMetrics
+
+/// A Swift Metrics `MetricsFactory` implementation backed by ``OTelMetricRegistry``.
+///
+/// - TODO: Review the API surface we want to provide.
+///
+///   Right now, this is a wrapper type around the ``OTelMetricRegistry`` which adds some configuration. The API is
+///   similar to what is in Swift Prometheus, but we might not need to layer this way.
+///
+///   Specifically, Swift Prometheus provides public API on the registry itself for directly creating and exporting
+///   metrics, without using the Swift Metrics API.
+///
+///   Currently, the registry in this package is opaque, which begs the question: maybe the registry type should be
+///   internal and we just have this factory be the public API for now and defer making the registry public to if/when
+///   we want to extend this package to provide direct OTel funcionality.
+public struct OTLPMetricsFactory: Sendable {
```

Comment:
I'm calling this out here and have also added it to the open questions section on the covering issue.

One thing we might like to consider, during this period, is we make all possibly-public types `@_spi(ExperimentalMetrics)`. This would allow you to keep this repo release-able during any bikeshedding of the public API for metrics, e.g. if you have a non-metrics related fix come in, for which you need to cut a release. 

WDYT @slashmo @ktoso

Reply:
Yeah, I think that's a good idea. I haven't tagged a 1.0, so technically, we don't _have_ to do it, but I guess it's not too much effort to add this to a couple of types.

Reply:
OK, I added a commit to this PR to make the types introduced SPI (including the registry, which was introduced in the previous PR).